contracts/facets/ViewStateFacet.sol

```diff
@@ -242,54 +212,61 @@ contract ViewStateFacet {
         return statsList;
     }
 
-    /// @notice Returns a sarcophagus.
-    /// @param sarcoId The identifier of the sarcophagus being returned
-    function getSarcophagus(bytes32 sarcoId)
-        external
-        view
-        returns (LibTypes.Sarcophagus memory)
-    {
-        LibTypes.Sarcophagus memory sarco = s.sarcophagi[sarcoId];
-        if (sarco.state != LibTypes.SarcophagusState.Active) {
-            return sarco;
-        }
-
-        uint8 unwrapsCount = 0;
-        for (uint8 i = 0; i < sarco.archaeologists.length; i++) {
-            bytes memory shard = s
-            .sarcophagusArchaeologists[sarcoId][sarco.archaeologists[i]]
-                .unencryptedShard;
+    struct SarcophagusResponse {
+        uint256 resurrectionTime;
+        bool isCompromised;
+        string name;
+        uint8 threshold;
+        uint256 maximumRewrapInterval;
+        string[2] arweaveTxIds;
+        address embalmerAddress;
+        address recipientAddress;
+        address[] archaeologistAddresses;
+        uint8 publishedKeyShareCount;
```

Comment:
We're not able to return the entire sarcophagus object (with its mapping of cursed archaeologists) because we can't return a dynamic type from a view function. Instead, I added two aggregate properties `publishedKeyShareCount` and `hasLockedBond` on the return type so we'll have the data we need for the dashboard with only one request.